src/scheduling_algorithms.c

```diff
@@ -19,136 +20,79 @@ void resetCurrentQuantum(SchedulerConfig *schedulerConfig)
+        // TODO:
+        // 1. Send Sleep Signal To Running Process.
+
+        // 2. Push Process To The Back Of The Queue.
+        push(head, running_process->process, running_process->priority);
+        pop(head);
+        // 3. Reset Quantum.
+        resetCurrentQuantum(schedulerConfig);
     }
 }
 
 /**
- * scheduleSRTF - Runs the shortest remaining time first algorithm
+ * scheduleSRTN - Runs the shortest remaining time first algorithm
  *
  * @param head: Pointer to the pointer to the head of the priority queue.
- * @param current_process: Pointer to the current running process by the cpu.
  *
- * Description: checks for null value for queue or the head of the queue.
- *              if their is no running process it gets one from the queue.
- *              else if a new process came but with less remaining time "which is the priority here" we switch between them and push the current to the queue again.
- *              update the priority of the current process as it represent the remaining time of the process.
- *              checks for the remaining time to send a termination signal to the schedular.
+ * Description: Checks if their is a running process and decrease its priority
+ *              as it indicates its remaining running time.
  */
-void scheduleSRTN(pqueue_t **head, rprocess_t *current_process)
+void scheduleSRTN(pqueue_t **head)
 {
+    pqueue_t *running_process;
     if (head == NULL)
     {
         fprintf(stderr, "Can't pop an empty queue\n");
         exit(EXIT_FAILURE);
     }
-
-    if (*head == NULL)
+    running_process = *head;
+    if (running_process == NULL)
         return;
 
-    if (current_process == NULL)
-        updateCurrentProcess(head, current_process);
-    else if ((*head)->priority < current_process->priority)
-    {
-        push(head, current_process->process, current_process->priority);
-        updateCurrentProcess(head, current_process);
-    }
-
-    current_process->priority--;
-    decrementRemainingCPUTime(current_process->process);
-
-    if (current_process->process->runtime == 0)
-    {
-        current_process = NULL;
-        // TODO: send termination signal.
-    }
+    // Decrease The Running Process Runtime.
+    running_process->priority--;
 }
 
 /**
  * scheduleHPF - Schedule a process using the Highest Priority First (HPF) algorithm (non-preemptive).
  * @param head: Pointer to the head of the priority queue.
- * @param current_process: Pointer to the current process being scheduled.
  *
- * Description: Implements scheduling logic for HPF algorithm (non-preemptive).
- *              Decrements current process's remaining CPU time, updates if finished, and handles edge cases.
+ * Description: Useless for now
  */
-void scheduleHPF(pqueue_t **head, rprocess_t *current_process)
+void scheduleHPF(pqueue_t **head)
 {
     if (head == NULL)
     {
         fprintf(stderr, "Can't pop an empty queue\n");
         exit(EXIT_FAILURE);
     }
```

Comment:
Note that there would me moments where queue is empty, but that is totally normal and not an errror

Reply:
The queue can be empty but the pointer to the queue must be allocated. The other case is if i exit on ```*head == null```